daemon/userconsent_test.go

```diff
@@ -0,0 +1,323 @@
+//go:build moose
+
+package daemon
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/NordSecurity/nordvpn-linux/auth"
+	"github.com/NordSecurity/nordvpn-linux/config"
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/test/category"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestModeForCountryCode(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	tests := []struct {
+		code     countryCode
+		expected consentMode
+	}{
+		{countryCode("us"), consentModeStandard},
+		{countryCode("ca"), consentModeStandard},
+		{countryCode("jp"), consentModeStandard},
+		{countryCode("au"), consentModeStandard},
+		{countryCode("fr"), consentModeGDPR},
+		{countryCode("zz"), consentModeGDPR},
+		{countryCode(""), consentModeGDPR},
+	}
+
+	for _, tt := range tests {
+		t.Run(string(tt.code), func(t *testing.T) {
+			mode := modeForCountryCode(tt.code)
+			assert.Equal(t, mode, tt.expected)
+		})
+	}
+}
+
+func TestModeForCountryCode_CaseInsensitive(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	codes := []string{"US", "Us", "uS", "us"}
+	for _, codeStr := range codes {
+		t.Run(codeStr, func(t *testing.T) {
+			cc := countryCode(strings.ToLower(codeStr))
```

Comment:
This makes the test invalid as it works because `strings.ToLower()` is called in the test and not in the lookup function.